keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/PausedRepository.kt

```diff
@@ -18,15 +18,18 @@
 package com.netflix.spinnaker.keel.persistence
 
 /**
- * Stores which applications have opted out of resource checking
+ * A repository to track what scopes are paused, starting with application
  */
-interface ApplicationVetoRepository {
+interface PausedRepository {
```

Comment:
Updating this repository (removing it, creating a new one) so that we can pause by scope other than application - like by environment, or delivery config.

When we roll this out we will have to note if any apps are currently paused and re-pause them manually because I did not write a migration.

Reply:
@emjburns Liked the refactoring from veto to paused - it makes it much easier to understand.

Reply:
Agreed

Reply:
I like the refactoring too, but I feel like `PausedRepository` is a bit weird/ambiguous as a name. Maybe `PausedScopeRepository`, or `PausedObjectRepository`, or even `PauseTrackingRepository`?

Reply:
Or maybe double down and name it `Pausitory`?

Reply:
Ohhhh, I want to name it `Pausitory` so so sooooo bad.

Reply:
On a more serious note, I felt like since every repository has to do with resources it makes the names too long and similar to have everything be "xResourceRepository". That's why I chose "PausedRepository".

Reply:
FWIW, one of the reasons I suggested the names I did was that, as per your PR, there are other 
"scopes" that can be paused beyond resources (e.g. applications). In other words, no `Resource` in my suggested names either. 😃